app/containers/ItunesContainer/tests/index.test.js

```diff
@@ -53,7 +63,47 @@ describe('ItunesContainer Tests', () => {
       resultCount: 50,
       results: [{ songName: 'Song1' }, { songName: 'Song2' }]
     };
-    const { getByTestId } = renderProvider(<ItunesContainer gridData={gridData} />);
+    const { getByTestId } = renderProvider(<ItunesWrapper gridData={gridData} />);
     expect(getByTestId('for')).toBeInTheDocument();
   });
+
+  it('should dispatchSearchSongs on load, if searchTerm is already persisted and avaiable', async () => {
+    const searchTerm = 'Justin Beiber';
+    renderProvider(<ItunesWrapper searchTerm={searchTerm} dispatchSearchSongs={mockDispatchSearch} />);
+    await timeout(500);
+    expect(mockDispatchSearch).toBeCalled();
+  });
+
+  it('should validate mapDispatchToProps actions', async () => {
+    const dispatchSearchSpy = jest.fn();
+    const searchTerm = 'Justin Beiber';
+    const actions = {
+      dispatchSearchSongs: { searchTerm, type: 'SEARCH_ITUNES' },
+      dispatchClearGridData: { type: 'CLEAR_GRID_DATA' }
+    };
+
+    const props = mapDispatchToProps(dispatchSearchSpy);
+    props.dispatchSearchSongs(searchTerm);
+    expect(dispatchSearchSpy).toHaveBeenCalledWith(actions.dispatchSearchSongs);
+
+    await timeout(500);
+    props.dispatchClearGridData();
+    expect(dispatchSearchSpy).toHaveBeenCalledWith(actions.dispatchClearGridData);
+  });
+
+  it('should render the error message', async () => {
+    const { getByTestId } = renderProvider(
+      <ItunesWrapper dispatchSearchSongs={mockDispatchSearch} searchError="error" />
+    );
+
+    await timeout(500);
+    expect(getByTestId('itunes-error-message')).toBeInTheDocument();
+  });
+
+  it('should render the default message', async () => {
```

Comment:
write a test to make sure that if 50 tracks are passsed, 50 trackCards are rendered

Reply:
Done